lib/src/analyzers/lint_analyzer/rules/rules_list/avoid_creating_vector_in_update/avoid_creating_vector_in_update_rule.dart

```diff
@@ -0,0 +1,46 @@
+// ignore_for_file: public_member_api_docs
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+// ignore: implementation_imports
+import 'package:collection/collection.dart';
+
+import '../../../../../utils/flame_type_utils.dart';
+import '../../../../../utils/node_utils.dart';
+import '../../../lint_utils.dart';
+import '../../../models/internal_resolved_unit_result.dart';
+import '../../../models/issue.dart';
+import '../../../models/severity.dart';
+import '../../models/flame_rule.dart';
+import '../../rule_utils.dart';
+
+part 'visitor.dart';
+
+class AvoidCreatingVectorInUpdateRule extends FlameRule {
+  static const String ruleId = 'avoid-creating-vector-in-update';
+
+  static const _warningMessage = "Avoid creating Vector2 in 'update' method.";
```

Comment:
@incendial why we support only Vector2? Please take a look https://pub.dev/documentation/flame/latest/input/Vector-class.html